packages/web-components/src/components/horizontal-rule/horizontal-rule.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, LitElement } from 'lit-element';
+// import ifNonNull from 'carbon-web-components/es/globals/directives/if-non-null';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import styles from './horizontal-rule.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Horizontal Rule
+ *
+ * @element dds-hr
+ */
+@customElement(`${ddsPrefix}-hr`)
+class DDSHorizontalRule extends LitElement {
+  /**
+   * Style of the HR (solid (default) | dashed)
+   */
+  @property({ attribute: 'style' })
+  style = 'solid';
+
+  /**
+   * Length of the HR
+   * (fluid (default) | small | medium | large)
+   */
+  @property({ attribute: 'size' })
+  size = 'fluid';
+
+  /**
+   * Contrast of the HR
+   * (medium-contrast (default) | low-contrast | high-contrast)
+   */
+  @property({ attribute: 'contrast' })
+  contrast = 'medium-contrast';
+
+  /**
+   * Weight of the HR
+   * (thin (default) | thick)
+   */
+  @property({ attribute: 'weight' })
+  weight = 'thin';
+
+  render() {
+    const { style, size, contrast, weight } = this;
+    return html`
+      <hr
+        class="${prefix}--hr ${prefix}--hr--${style} ${prefix}--hr--${size} ${prefix}--hr--${contrast} ${prefix}--hr--${weight}"
+      />
```

Comment:
Unlike React, this creates `<hr>` in addition to the host element (`<dds-horizontal-rule>`), which will be two separate elements ([details](https://github.com/carbon-design-system/ibm-dotcom-library/blob/13d5159/packages/web-components/docs/coding-conventions.md#custom-element-itself-as-an-eleement)). To avoid that, we can add `reflect: true` to `@property()`s affecting the styles and use attribute selector to apply `.#{$prefix}--hr`, `.#{$prefix}--hr--solid`, etc., so that we can let `<dds-horizontal-rule>` itself works as a `<hr>`.